.babelrc

```diff
@@ -1,15 +1,28 @@
 {
-  "presets": ["next/babel"],
+  "presets": [
+    "next/babel"
+  ],
   "plugins": [
     [
       "module-resolver",
       {
-        "root": ["."],
+        "root": [
+          "."
+        ],
         "alias": {
           "client": "./client",
-          "server": "./server"
+          "server": "./server",
+          "styles": "./styles"
```

Comment:
will `styles/` ever be accessed from `server/`?  if not, can we put it in `client/`?

Reply:
Since `next.js` is taking care of SSR and our current setup has us grabbing components out of `client/` from `pages/` where the SSR magic happens, I think it makes sense for styles to also be in `client/`.  I haven't toyed around with any advanced config around SSR though so I'm not sure if we'd need to import anything from `styles/` to the `server/` directory. Would you happen to know of any instances where it would be needed to do such an import?

Reply:
@timmyichen Sorry - I've been swamped all day. I'm just seeing this. So the scenario I was talking about is when we need to **inline critical above-the-fold CSS** like mentioned [here](https://github.com/emotion-js/emotion/issues/1178#issuecomment-520493094). Lighthouse throws warnings when moderate sized CSS files are loaded via `link` as the import is render-blocking. An instance of this is if we are using a light enough Grid/Flex framework, it might make sense to inline it. 

Here are some instances where we do inlining: [1](https://github.com/zeit/next.js/blob/9c4eefcdbf4058b65c8b88fe98387166acad127b/examples/with-babel-macros/pages/_document.js#L9), [2](https://github.com/zeit/next.js/blob/9c4eefcdbf4058b65c8b88fe98387166acad127b/examples/with-cxs/pages/_document.js#L17), [3](https://github.com/zeit/next.js/blob/5782651e92ef07845e9d251981ac3ba2ced27e11/examples/with-fela/pages/_document.js#L27)

The way we are hooking up `styled-components` on [this PR](https://github.com/freeCodeCamp/chapter/pull/121/files#diff-8f3a5cf5019173ef1f56a54e45e44747R20) is also by inlining the styles although it isn't as straight forward as that.